### ISSUE TITLE
[Question] how to split train, validation, and test set

### ISSUE THREAD

**Issue by nevermet** [Lightning-AI/litgpt#683]
Dear all,

I have three 3 sets of data for train, validation, and test.

However,  when I run prepare_mydata.py, it seems to split the data into train and validation set randomly... So, I merged train and validation data set into one file and ran prepare_mydata.py. I will run generate.py against the finetuned model with my test data so that I can compare the models.

If I want to, can I specify train and validation to the data set when I run the prepare_mydata.py?

Thank you in advance.

**Comment by carmocca** [comment]
You can add a validation partition to this piece of code: https://github.com/Lightning-AI/lit-gpt/blob/main/scripts/prepare_alpaca.py#L50-L54
https://pytorch.org/docs/stable/data.html#torch.utils.data.random_split

**Comment by rasbt** [comment]
Closing since I think the question was addressed. But please feel free to reopen with any follow-up questions or concerns!
